dds_glossary/main.py

```diff
@@ -1,194 +1,48 @@
 """Main entry for the dds_glossary server."""
-from os import getenv
 
+import argparse
 from http import HTTPStatus
+from os import getenv
 
-from fastapi import Depends, FastAPI, Request
-from fastapi.responses import JSONResponse
-from fastapi.templating import Jinja2Templates
 import sentry_sdk
-from starlette.templating import _TemplateResponse
+import uvicorn
+from fastapi import FastAPI
+from fastapi.templating import Jinja2Templates
 
 from . import __version__
-from .auth import get_api_key
 from .controllers import GlossaryController
+from .routes import router
 
 
-SENTRY_DSN=getenv("SENTRY_DSN")
-
-sentry_sdk.init(
-    dsn=SENTRY_DSN,
-    # Set traces_sample_rate to 1.0 to capture 100%
-    # of transactions for performance monitoring.
-    traces_sample_rate=1.0,
-    # Set profiles_sample_rate to 1.0 to profile 100%
-    # of sampled transactions.
-    # We recommend adjusting this value in production.
-    profiles_sample_rate=1.0,
-)
-
-app = FastAPI()
-controller = GlossaryController()
-templates = Jinja2Templates(directory="templates")
-
-
-@app.get("/")
-def home(
-    request: Request,
-    search_term: str = "",
-    concept_scheme_iri: str = "",
-    lang: str = "en",
-) -> _TemplateResponse:
-    """Get the home page.
-    If a `search_term` term is provided, it will filter the concepts by the search term.
-    If a `concept_scheme_iri` is provided, it will filter the concepts by the concept
-    scheme.
-
-    Args:
-        request (Request): The request.
-        search_term (str): The search term to filter the concepts. Defaults to "".
-        concept_scheme_iri (str): The concept scheme IRI to filter the concepts.
-            Defaults to "".
-        lang (str): The language to use for searching concepts. Defaults to "en".
-
+def create_app() -> FastAPI:
+    """Create the FastAPI application object
     Returns:
-        _TemplateResponse: The home page with search results if any.
+        FastAPI: the application object
     """
-    if concept_scheme_iri:
-        concepts = controller.get_concepts(concept_scheme_iri, lang=lang)
-    else:
-        concepts = controller.search_database(search_term, lang=lang)
-
-    return templates.TemplateResponse(
-        "home.html",
-        {
-            "request": request,
-            "schemes": controller.get_concept_schemes(lang=lang),
-            "concepts": concepts,
-        },
-    )
-
-@app.get("/search")
-def search(
-    search_term: str ,
-    lang: str = "en",
-) -> JSONResponse:
-    """Search concepts according to given expression.
-    Note: This will be removed once #35 (Add elasticsearch) is closed.
 
-    Args:
-        search_term (str): The search term to filter the concepts.
-        lang (str): The language to use for searching concepts. Defaults to "en".
+    SENTRY_DSN = getenv("SENTRY_DSN")
```

Comment:
@jsvgoncalves we should add error handling for this.

Reply:
won't really care now, since we'll https://github.com/Depart-de-Sentier/dds_glossary/issues/37